atef/procedure.py

```diff
@@ -81,27 +159,142 @@ class DeviceConfiguration:
 class ConfigurationCheckStep(ProcedureStep):
     """Step which checks device configuration versus a given timestamp."""
     #: Device name to device configuration information.
-    devices: Dict[str, DeviceConfiguration]
+    devices: Dict[str, DeviceConfiguration] = field(default_factory=dict)
 
 
 @dataclass
 class TyphosDisplayStep(ProcedureStep):
     """A procedure step which opens one or more typhos displays."""
     #: Happi device name to display options.
-    devices: Dict[str, DisplayOptions]
+    devices: Dict[str, DisplayOptions] = field(default_factory=dict)
 
 
 @dataclass
 class PydmDisplayStep(ProcedureStep):
     """A procedure step which a opens a PyDM display."""
     #: The display path.
-    display: pathlib.Path
+    display: pathlib.Path = field(default_factory=pathlib.Path)
     #: Options for displaying.
-    options: DisplayOptions
+    options: DisplayOptions = field(default_factory=DisplayOptions)
+
+
+@dataclass
+class PassiveStep(ProcedureStep):
+    """A step that runs a passive checkout file"""
+    filepath: pathlib.Path = field(default_factory=pathlib.Path)
+
+    def _run(self) -> Result:
+        """ Load, prepare, and run the passive step """
+        if self.filepath.suffix == '.json':
+            config = ConfigurationFile.from_json(self.filepath)
+        else:
+            config = ConfigurationFile.from_yaml(self.filepath)
+
+        # prepare file
+        prepared_config = PreparedFile.from_config(file=config,
+                                                   cache=DataCache())
+
+        # run passive checkout.  Will need to set up asyncio loop
+        loop = asyncio.get_event_loop()
+        coroutine = run_passive_step(prepared_config)
+        result = loop.run_until_complete(coroutine)
+
+        return result
 
 
 @dataclass
 class ProcedureGroup(ProcedureStep):
     """A group of procedure steps (or nested groups)."""
     #: Steps included in the procedure.
-    steps: Sequence[Union[ProcedureStep, ProcedureGroup]]
+    steps: Sequence[Union[ProcedureStep, ProcedureGroup]] = \
```

Comment:
Is the `\` continuation really required with our flake8 settings? I'd think probably not?

Reply:
It's actually improper syntax to split the line there without the `\`

Reply:
I meant that the line splitting itself might not have been required

Reply:
aaaah right that's fair.  I'm starting to feel like I'm more strict than flake8 on line length...